bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/progress/UIJob.java

```diff
@@ -31,6 +34,53 @@
  * @since 3.0
  */
 public abstract class UIJob extends Job {
+
+	/**
+	 * Creates a new UIJob that will execute the provided function in the UI thread
+	 * when it runs.
+	 *
+	 * Prefer using {@link UIJob#createUI(String, ICoreRunnable)}.
+	 *
+	 * @param name     The name of the job
+	 * @param function The function to execute
+	 * @return A UIJob that encapsulates the provided function
+	 * @see IJobFunction
+	 * @since 3.127
+	 */
+	public static UIJob createUI(String name, IJobFunction function) {
```

Comment:
why "createUI" and not simply "create"? We don't create widgets here :)

Reply:
> why "createUI" and not simply "create"? We don't create widgets here :)

That's right. 😅 The super-class Job has `create()` methods that would clash with this class.
For example if one calls `UIJob.create()` now (without this change) a Job is returned. If we add just a create() method that would now return a `UIJob`. A UIJob is still a Job, so that would not be a compiler-error, but it would suddenly execute in the UI-thread, instead of a worker thread.
One can probably assume that that was the actual intention, nevertheless it would be a behavioral change.

The name createUI is indeed misleading and another would be better (maybe `createUIJob()`?), but just `create()` could cause problems.

Reply:
I don't think you can override static methods...

Reply:
Indeed you can't, but you can hide the static base class methods by defining a method with the same name in the subclass.  Note however that if one did write UIJob.create that there would be a warning `The static method create(String, ICoreRunnable) from the type Job should be accessed directly` unless one turns off such a warning.   I suppose it's possible that some consumer ignored that warning or turned it off, but that seems somewhat unlikely.

Reply:
Yes, that's what I meant and yes precisely it is is not a override.

The question to me is if we want to take that risk or not? As you said there is a warning, so one could argue that it is the users fault if that warning is ignored.
And yes, just `UIJob.create()` would definitely be nicer.

Reply:
I think anyone writing UIJob.create would expect it to create a UIJob and although it doesn't currently do that, changing it now is at worst a source incompatibility not a binary incompatibility.  If someone was affected by the change, they could use Job.create instead as they ought to have done in the first place.  That would still be binary compatible with their intent in both an old Eclipse and a newer one.  So personally I'd be tempted to do what's nicer as an API...

Reply:
> I think anyone writing UIJob.create would expect it to create a UIJob and although it doesn't currently do that, changing it now is at worst a source incompatibility not a binary incompatibility. 

I think the same. And it is not even a source incompatibility because before UIJob.create() returned a Job and now it returns an `UIJob`, which is still a job. So unless somebody notices less warnings the only (maybe) noticeable effect is that the Job now runs in the UI-thread. That was probably intended in the first place, however it still changed now.

I'm fine with that change. From my experience I just noticed that the Eclipse-community (or maybe consumers in general) is sensitive to changes. Therefore I wanted to raise awareness about this point and if other devs with more experience, like you, say it is OK, I'm fine with that as well. :)